js/ui/components/clubhouse.js

```diff
@@ -564,13 +565,28 @@ var ClubhouseNotificationBanner = new Lang.Class({
                          { x: endX,
                            time: CLUBHOUSE_BANNER_ANIMATION_TIME,
                            transition: 'easeOutQuad',
+                           onUpdate: this._clipActor.bind(this),
                            onComplete: () => {
                                this.actor.destroy();
                                this.actor = null;
                            }
                          });
     },
 
+    _clipActor: function() {
+        let monitor = Main.layoutManager.primaryMonitor;
+        if (!monitor) {
+            return;
+        }
+
+        let monitor_edge = monitor.x + monitor.width;
+        let actor_edge = this.actor.x + this.actor.width;
+        let offset = actor_edge - monitor_edge;
+        offset = offset > 0 ? offset : 0;
```

Comment:
You can simplify this with `let offset = Math.max(0, actor_edge - monitor_edge);`